### DISCUSSION TITLE
Copy/paste styles between components

### DISCUSSION THREAD

**Discussion by havrylenko-olena** [GrapesJS/grapesjs#4539] [Ideas]
It would be nice to have ability to copy/paste styles between grapesjs components. 
Maybe like some additional buttons on Styles panel.

**Comment by AntonMelnykAA** [comment]
+

**Comment by lexoyo** [comment]
I may have missunderstood but isn't css classes what you are looking for?

**Reply by havrylenko-olena** [reply]
Not exactly. I mean situation when you need to copy/paste some styles between components not related to classes, or maybe some classes have to have some same styles.

**Comment by artf** [comment]
I'm not sure about adding something new to the core UI but you should be able to implement that functionality relatively simply by using the StyleManager API.
```js
// Using for convenience a simple variable to keep track of copied styles
let copiedStyles = {};
const { StyleManager, Commands } = editor;

Commands.add('copy-styles', () => {
  copiedStyles = StyleManager.getSelected()?.getStyle() || {};
});

Commands.add('paste-styles', () => {
  StyleManager.getSelected()?.addStyle(copiedStyles);
});
```
Using the Commands API is not mandatory, but that might help to make this functionality more reusable if you decide to place it in a plugin  (eg. to use commands `editor.runCommand('copy-styles')`/`editor.runCommand('paste-styles')`)

**Reply by havrylenko-olena** [reply]
Thanks, @artf
